src/libsync/creds/oauth.cpp

```diff
@@ -576,12 +620,19 @@ void AccountBasedOAuth::fetchWellKnown()
 void AccountBasedOAuth::refreshAuthentication(const QString &refreshToken)
 {
     if (!OC_ENSURE(!_isRefreshingToken)) {
+        qCDebug(lcOauth) << "already refreshing token, aborting";
         return;
     }
+
     _isRefreshingToken = true;
+
+    qCDebug(lcOauth) << "fetching dynamic registration data";
+
     auto credentialsJob = _account->credentialManager()->get(dynamicRegistrationDataC());
 
-    connect(credentialsJob, &CredentialJob::finished, this, [=] {
+    connect(credentialsJob, &CredentialJob::finished, this, [this, credentialsJob, refreshToken] {
```

Comment:
I prefer, to put this last, but we should come up with a standard on how to capture this.